### ISSUE TITLE
Upgrading to wagtail 1.6 breaks, with DEBUG=False and ManifestStaticFilesStorage as STATICFILES_STORAGE

### ISSUE THREAD

**Issue by mevinbabuc** [torchbox/wagtail#2927]
After upgrading to wagtail 1.6 wagtail admin fails to load a bunch of JS scripts.  None of the files specified in `wagtailadmin/templates/wagtailadmin/pages/_editor_js.html` is rendered on to the edit or create page HTML.

Found that this Django setting was causing it

```python
DEBUG=False
STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.ManifestStaticFilesStorage'
```

Switching to a normal Django static storage works. but we would like to fingerprint our assets. This also prevents us from using custom static files storage like whitenoise.  This was not an issue with the previous versions of wagtail (1.5.3).

**Setup**
* Python 3.4
* Django 1.8.13
* Wagtail 1.6

**Steps to reproduce**
```shell
pip install django==1.8
wagtail start clean
python3 manage.py migrate
```

Edit the settings to 
>
```python
DEBUG=False
STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.ManifestStaticFilesStorage'
```

```shell
python manage.py collectstatic --clear
python manage.py runserver 8000
```

@gasman @alexgleason @m1kola can you guys take a look ?

**Comment by gasman** [comment]
`manage.py runserver` does not serve static files when `DEBUG=False` (see https://docs.djangoproject.com/en/1.10/howto/static-files/#serving-static-files-during-development), so your reproduction steps will fail on any version of Django / Wagtail. On a production server, you should configure your webserver to serve static files directly (i.e. not going through Django): https://docs.djangoproject.com/en/1.10/howto/static-files/deployment/

With further hacks (removing the `if settings.DEBUG` check from urls.py, and running `manage.py runserver --insecure`) you can override the `runserver` behaviour and get it to serve static files again - however, in this case it'll take them from their original locations in the app directories, not from the collected files in STATIC_ROOT. I suspect this is why you're having problems with ManifestStaticFilesStorage: the `{% static %}` tags are correctly adding the hash suffix to the filename, but `runserver` only knows about the un-suffixed files.

If you can find a set of steps that is confirmed to work on Wagtail <=1.5 and fails on >1.6, we can take another look...

**Comment by mevinbabuc** [comment]
It's failing in production and hence I have opened the ticket.
The static files are served via nginx. I would like if you could try to run
wagtail 1.6 with the specified storage settings which I specified above.
It works flawlessly with wagtail 1.5.

Accoring to ticket the bug is that the js specified in _editor_js.html is
failing to appear in the rendered HTML code.

**Comment by gasman** [comment]
OK, I think I understand the problem now! I missed your comment about _editor_js.html not rendering at all, as that wasn't in the original report that I received by email. (Please don't rely on Github's 'edit comment' feature to add substantial new information - that causes miscommunications...)

In Wagtail 1.6, we added the file `admin/js/vendor/xregexp/xregexp.min.js` to _editor_js.html. This file is part of django.contrib.admin on Django 1.9 and above; on Django 1.8 (or when django.contrib.admin is not listed in INSTALLED_APPS), it's a missing file. Normally, this would be silently ignored - unicode slug support is disabled but everything else works fine - but ManifestStaticFilesStorage breaks on non-existent files, causing the whole _editor_js.html include to be skipped.

**Comment by mevinbabuc** [comment]
Moving django is not an option because it's not an LTS version. Will
placing a copy of xregexp.min.js fix the issue or is there any work around
that you suggest ?

**Comment by mevinbabuc** [comment]
Will keeping a copy of admin/js/vendor/xregexp/xregexp.min.js resolve this ? What do you suggest as a fix ? maybe I can raise a PR for this, if you have any ideas on how this can be resolved.

How is xregexp.min.js helping Wagtail ? Just want to understand more about it, if its affecting any wagtail features.

**Comment by mevinbabuc** [comment]
Will adding a template tag to check django version and loading the JS file if its greater than or equal to  1.9 work ? 

```python

@register.simple_tag
def is_django_1_9():
    if django.VERSION >= (1, 9, 0):
        return True
    else:
        return False

```

**Comment by gasman** [comment]
As an immediate workaround, creating a _blank_ `admin/js/vendor/xregexp/xregexp.min.js` file in one of your app's `static` directories would probably work. That way, _editor_js.html will render correctly, but the JS code will still fall back on the non-Unicode Django 1.8 version, as it would if xregexp.min.js was a 404.

**Comment by vishnuratheesh** [comment]
the workaround - worked!

**Comment by m1kola** [comment]
Fixed in #2937

Thanks for reporting!
